Sources/SwiftKuery/ConnectionPool.swift

```diff
@@ -125,6 +125,15 @@ public class ConnectionPool {
         }
         item = pool[0]
         pool.removeFirst()
+        // Check if the item is alive, and append a new one to the pool if it isn't
+        if item.isConnected == false {
+            releaser(item)
+            capacity += 1
```

Comment:
I don't think this is right. Does `releaser` decrement the capacity? Otherwise we are just replacing a dead connection with a fresh one.
If we do need to increment, it should be within the `if let` below.

Reply:
`Releaser` doesn't decrement it, I've moved the capacity change to within the `if let`.